src/SprykerEco/Zed/Braintree/Business/Hook/PostSaveHook.php

```diff
@@ -36,6 +36,10 @@ public function __construct(BraintreeRepositoryInterface $repository)
      */
     public function postSaveHook(QuoteTransfer $quoteTransfer, CheckoutResponseTransfer $checkoutResponse)
     {
+        if (!$checkoutResponse->getSaveOrder()->getIdSalesOrder()) {
```

Comment:
you have to use either orFail if it was required before
or check`getSaveOrder` existence first

Reply:
Since initially there were no checks for non-empty `$checkoutResponse->getSaveOrder()`  and the plugin would fail anyway for an empty value just a few lines down this place, using OrFail would not lead to a BC break. Changed to OrFail.